shell.nix

```diff
@@ -39,8 +39,8 @@ pkgs.stdenv.mkDerivation {
     z3
   ];
   shellHook = ''
-    export PATH=$PWD/node_modules/.bin/:$PWD/bin:$PATH
-    export KLAB_EVMS_PATH="''${KLAB_EVMS_PATH:-''${PWD}/evm-semantics}"
+    export PATH=${toString ./node_modules/.bin}:${toString ./bin}:$PATH
+    export KLAB_EVMS_PATH="''${KLAB_EVMS_PATH:-${toString ./evm-semantics}}"
```

Comment:
Do we really need to do this default-argument thing?

If this file is sourced directly, we can set the var to its desired value. If we're depending upon klab in a parent project, the variable will be assigned to in the parent's `shell.nix` (and this one will not be sourced).

Are there cases where we source this file while having already set this variable, and are they legitimate?

I'm also happy to do it in another PR if you prefer.

Reply:
> Are there cases where we source this file while having already set this variable, and are they legitimate?

Good question. I guess the use case would be to quickly try out different `evms` versions with the same `klab`, but that can also be achieved by checking out branches in the submodule.

I think if we remove the default arg stuff here we should consider removing the variable entirely for the same reason?

Reply:
> Good question. I guess the use case would be to quickly try out different evms versions with the same klab, but that can also be achieved by checking out branches in the submodule.

Yeah, so far I've used git (branches and worktrees) to test out different versions of evm/klab/whatever. That's a bit more convoluted but it's also more general and doesn't require special support.

> I think if we remove the default arg stuff here we should consider hardcoding the value within klab itself for the same reason?

Yeah good point. We could remove both, unless keeping it would make more sense for non-nix users, of which I'm not sure.

Let's move this to a new PR, since touching klab proper would be out of scope.

Reply:
> Let's move this to a new PR, since touching klab proper would be out of scope.

:+1: